src/main/java/io/github/intoto/helpers/IntotoHelper.java

```diff
@@ -0,0 +1,158 @@
+   * @return a JSON representation for the envelope.
+   * @throws InvalidModelException thrown when the given statement is not valid
+   * @throws JsonProcessingException thrown when there are issues generating the JSON string
+   * @throws NoSuchAlgorithmException thrown when there are issues encrypting the payloads in the
+   *     Envelope
+   * @throws SignatureException thrown when there are issues with the given key in the Signer
+   * @throws InvalidKeyException thrown when there are issues matching the key with the given
+   *     algorithm
+   */
+  public static String produceIntotoEnvelopeAsJson(
+      Statement statement, Signer signer, boolean prettyPrint)
+      throws InvalidModelException, JsonProcessingException, NoSuchAlgorithmException,
+          SignatureException, InvalidKeyException {
+    // Get the Base64 encoded Statement to use as the payload
+    String jsonStatement = validateAndTransformToJson(statement, false);
+    String base64EncodedStatement = Base64.getEncoder().encodeToString(jsonStatement.getBytes());
+
+    IntotoEnvelope envelope = new IntotoEnvelope();
+    // Create the signed payload with the DSSEv1 format and sign it!
+    byte[] signedDsseV1Payload =
+        signer.sign(
+            createPreAuthenticationEncoding(envelope.getPayloadType(), base64EncodedStatement));
+
+    Signature signature = new Signature();
+    signature.setKeyId(signer.getKeyId());
+    // The sig contains the base64 encoded version of the signedDsseV1Payload
+    signature.setSig(Base64.getEncoder().encodeToString(signedDsseV1Payload));
+    // Let's complete the envelope
+    envelope.setPayload(base64EncodedStatement);
+    envelope.setSignatures(List.of(signature));
+    if (prettyPrint) {
+      return objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(envelope);
+    }
+    return objectMapper.writeValueAsString(envelope);
+  }
+
+  /**
+   * Produces an {@link IntotoEnvelope} and signs the payload with the given Signer. Note: There is
+   * another convenience method that returns the serialized JSON representation for the envelope
+   *
+   * @param statement the Statement to add to the envelope
+   * @param signer the Signer that will be used to sign the payloads.
+   * @return will return a {@link IntotoEnvelope} instead of the JSON representation.
+   * @throws InvalidModelException thrown when the given statement is not valid
+   * @throws JsonProcessingException thrown when there are issues generating the JSON string
+   * @throws NoSuchAlgorithmException thrown when there are issues encrypting the payloads in the *
+   *     Envelope
+   * @throws SignatureException thrown when there are issues with the given key in the Signer
+   * @throws InvalidKeyException thrown when there are issues matching the key with the given *
+   *     algorithm
+   */
+  public static IntotoEnvelope produceIntotoEnvelope(Statement statement, Signer signer)
```

Comment:
Instead of having produceIntotoEnvelope and productIntotoEnvelopeAsJson duplicate each other's logic, why not have productIntotoEnvelopeAsJson _call_ produceIntotoEnvelope and then just convert the result to JSON?

Reply:
Done.